com.jetbrains.PyCharm-Professional.yaml

```diff
@@ -0,0 +1,134 @@
+        sha256: a5258b84f76661d558492fa87e42db634df143685a0e51802d59cae7daad8732
+      - type: file
+        url: https://files.pythonhosted.org/packages/ef/80/cef14194e2dd62582cc0a4f5f2db78fb00de3ba5d1bc0e50897b398ea984/invoke-1.2.0.tar.gz
+        sha256: dc492f8f17a0746e92081aec3f86ae0b4750bf41607ea2ad87e5a7b5705121b7
+      - type: file
+        url: https://files.pythonhosted.org/packages/74/75/ee4710c824c601a9201b71c6ee38b09d38f2db1796dabb193f3f80a4a9f5/parver-0.2.1.tar.gz
+        sha256: 1b37a691af145a3a193eff269d53ba5b2ab16dfbb65d47d85360755919f5fe4b
+      - type: file
+        url: https://files.pythonhosted.org/packages/1d/64/a18a487b4391a05b9c7f938b94a16d80305bf0369c6b0b9509e86165e1d3/setuptools-41.0.1.zip
+        sha256: a222d126f5471598053c9a77f4b5d4f26eaa1f150ad6e01dcf1a42e185d05613
+      - type: file
+        url: https://files.pythonhosted.org/packages/1d/b0/f478e80aeace42fe251225a86752799174a94314c4a80ebfc5bf0ab1153a/wheel-0.33.4.tar.gz
+        sha256: 62fcfa03d45b5b722539ccbc07b190e4bfff4bb9e3a4d470dd9f6a0981002565
+      - type: file
+        url: https://files.pythonhosted.org/packages/d7/a7/08b88808c409722361459f1ae24474530d83593d6ded346f1d3649326838/virtualenv-clone-0.5.3.tar.gz
+        sha256: c88ae171a11b087ea2513f260cdac9232461d8e9369bcd1dc143fc399d220557
+      - type: file
+        url: https://files.pythonhosted.org/packages/06/b8/d1ea38513c22e8c906275d135818fee16ad8495985956a9b7e2bb21942a1/certifi-2019.3.9.tar.gz
+        sha256: b26104d6835d1f5e49452a26eb2ff87fe7090b89dfcaee5ea2212697e1e1d7ae
+
+  - name: pycharm
+    buildsystem: simple
+    build-commands:
+      - tar xzfv pycharm-professional-2019.2.tar.gz
+      - rm -f pycharm-2019.2/help/*
+      - install -dm755 ${FLATPAK_DEST}/bin
+      - install -dm755 ${FLATPAK_DEST}/pycharm
+      - cp -r pycharm-2019.2/* ${FLATPAK_DEST}/pycharm
+      - chmod -R a-s,go+rX,go-w ${FLATPAK_DEST}/pycharm
+      - python3 ${FLATPAK_DEST}/pycharm/helpers/pydev/setup_cython.py build_ext --inplace
+      - install pycharm-2019.2/bin/pycharm.sh ${FLATPAK_DEST}/bin/pycharm
+      - install -t ${FLATPAK_DEST}/share/icons/hicolor/scalable/apps -Dm644 pycharm-2019.2/bin/pycharm.svg
+      - install -t ${FLATPAK_DEST}/share/applications/ -Dm644 com.jetbrains.PyCharm-Professional.desktop
+      - install -t ${FLATPAK_DEST}/share/appdata/ -Dm644 com.jetbrains.PyCharm-Professional.appdata.xml
+      - install -m755 pycharm-desktop ${FLATPAK_DEST}/bin/pycharm-desktop
+      - icon_in="${FLATPAK_DEST}/share/icons/hicolor/scalable/apps/pycharm.svg";
+        icon_out="${FLATPAK_ID}.png";
+        for s in {32,64,128,256,512}; do
+        rsvg-convert "${icon_in}" -w "${s}" -h "${s}" -a -f png -o "${icon_out}";
+        install -p -Dm644 "${icon_out}" -t "${FLATPAK_DEST}/share/icons/hicolor/${s}x${s}/apps/";
+        done;
+    sources:
+      - type: file
```

Comment:
This one should be `extra-data`...

Reply:
I figured could use archive instead of file and remove the `tar xvf`, but I don't quite understand why I should use extra-data. Could you elaborate please?

Reply:
Because Flathub cannot redistribute this app. extra-data will download the archive in the user machine during the install and install it which avoid redistribution issues.

Reply:
Do you have an example/doc I could check?

Reply:
We have a bunch of examples here on Flathub. Just search for extra-data on github.com/flathub.